python/astshim/fitsChanContinued.py

```diff
@@ -0,0 +1,258 @@
+from .fitsChan import FitsChan, CardType
+from .fitsChan import *  # noqa: F403 F401
+
+
+def _calc_card_pos(self, index):
+    """Convert a python index into a FitsChan position.
+
+    Parameters
+    ----------
+    self : `FitsChan`
+        The FitsChan to index.
+    index : `int`
+        0-based index into header. If negative, counts from end.
+    """
+    # Calculate 0-based index
+    if index < 0:
+        index = len(self) + index
+
+    # Convert to 1-based index
+    return index + 1
+
+
+def length(self):
+    return self.nCard
+
+
+FitsChan.__len__ = length
+
+
+def iter(self):
+    """The FitsChan itself is the iterator.
+    """
+    self.clearCard()
+    return self
+
+
+FitsChan.__iter__ = iter
+
+
+def next(self):
+    """Return the entire header card until we run out of cards.
```

Comment:
It returns the 80 character header card which is the fundamental unit of a FITS header.  This matches the standard convention of pyast.  Returning just the keys can't work because they can match multiple values and returning an integer index is superfluous. I'll see how it looks if I return a readonly dataclass FitsCard.

I would expect an `items()` method to return a tuple of keyword and value (which I can add).

Reply:
Thank you for the explanation. I agree the behavior is reasonable. Please document it in the doc string.

Adding an `items` method would be *really* helpful.